crates/release_plz_core/src/command/release.rs

```diff
@@ -170,18 +170,22 @@ impl ReleaseRequest {
         config.features.clone()
     }
 
-    fn registry_token(&self) -> Option<secrecy::Secret<String>> {
-        self.registry
-            .clone()
-            .and_then(|r| {
-                // Credentials for a specific registry can be set using environment variables.
-                // https://doc.rust-lang.org/cargo/reference/config.html#credentials
-                let env_token = env::var(format!("CARGO_REGISTRIES_{}_TOKEN", r.to_uppercase()))
-                    .ok()
-                    .map(SecretString::new);
-                self.token.clone().or(env_token)
-            })
-            .or(self.token.clone())
+    fn find_registry_token(
+        &self,
+        registry: &Option<String>,
+    ) -> anyhow::Result<Option<secrecy::Secret<String>>> {
+        if self.registry == *registry {
+            Ok(self
+                .token
+                .clone()
+                .or(cargo_utils::registry_token(&self.registry)?))
+        } else {
+            cargo_utils::registry_token(&self.registry)
+        }
+        .context(format!(
+            "can't retreive token for registry: {:?}",
```

Comment:
if registry is None this will print "can't retreive token for registry: None" which seems a bug

Reply:
yes I didn't know if we `None` or `crates.io`, idem about `Some('xxx')` or `'xxx'`. To have the info if value was defined or not.